airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/SecretsRepositoryWriter.java

```diff
@@ -0,0 +1,268 @@
+
+      // get all destination defs so that we can use their specs when storing secrets.
+      @SuppressWarnings("unchecked")
+      final List<StandardDestinationDefinition> destinationDefs =
+          (List<StandardDestinationDefinition>) augmentedMap.get(ConfigSchema.STANDARD_DESTINATION_DEFINITION).collect(Collectors.toList());
+      augmentedMap.put(ConfigSchema.STANDARD_DESTINATION_DEFINITION, destinationDefs.stream());
+      final Map<UUID, ConnectorSpecification> destinationDefIdToSpec = destinationDefs
+          .stream()
+          .collect(Collectors.toMap(StandardDestinationDefinition::getDestinationDefinitionId, StandardDestinationDefinition::getSpec));
+
+      if (augmentedMap.containsKey(ConfigSchema.SOURCE_CONNECTION)) {
+        final Stream<?> augmentedValue = augmentedMap.get(ConfigSchema.SOURCE_CONNECTION)
+            .map(config -> {
+              final SourceConnection source = (SourceConnection) config;
+
+              if (!sourceDefIdToSpec.containsKey(source.getSourceDefinitionId())) {
+                throw new RuntimeException(new ConfigNotFoundException(ConfigSchema.STANDARD_SOURCE_DEFINITION, source.getSourceDefinitionId()));
+              }
+
+              final var partialConfig = statefulSplitSecrets(
+                  source.getWorkspaceId(),
+                  source.getConfiguration(),
+                  sourceDefIdToSpec.get(source.getSourceDefinitionId()));
+
+              return source.withConfiguration(partialConfig);
+            });
+        augmentedMap.put(ConfigSchema.SOURCE_CONNECTION, augmentedValue);
+      }
+
+      if (augmentedMap.containsKey(ConfigSchema.DESTINATION_CONNECTION)) {
+        final Stream<?> augmentedValue = augmentedMap.get(ConfigSchema.DESTINATION_CONNECTION)
+            .map(config -> {
+              final DestinationConnection destination = (DestinationConnection) config;
+
+              if (!destinationDefIdToSpec.containsKey(destination.getDestinationDefinitionId())) {
+                throw new RuntimeException(
+                    new ConfigNotFoundException(ConfigSchema.STANDARD_DESTINATION_DEFINITION, destination.getDestinationDefinitionId()));
+              }
+
+              final var partialConfig = statefulSplitSecrets(
+                  destination.getWorkspaceId(),
+                  destination.getConfiguration(),
+                  destinationDefIdToSpec.get(destination.getDestinationDefinitionId()));
+
+              return destination.withConfiguration(partialConfig);
+            });
+        augmentedMap.put(ConfigSchema.DESTINATION_CONNECTION, augmentedValue);
+      }
+
+      configRepository.replaceAllConfigsNoSecrets(augmentedMap, dryRun);
+    } else {
+      configRepository.replaceAllConfigsNoSecrets(configs, dryRun);
+    }
+  }
+
+  public void loadData(final ConfigPersistence seedPersistence) throws IOException {
+    configRepository.loadDataNoSecrets(seedPersistence);
```

Comment:
Do we not need to split out any secrets here before calling `loadDataNoSecrets()`?

Reply:
good call. in practice, this doesn't matter. the seedPersistence will never have secrets in it. that said, it is misleading. i'm going to move this back into config repository to avoid confusion.